app/src/main/java/com/example/bigowlapp/utils/SupervisorSmsListener.java

```diff
@@ -0,0 +1,76 @@
+package com.example.bigowlapp.utils;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.telephony.SmsMessage;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.example.bigowlapp.model.Schedule;
+import com.google.firebase.Timestamp;
+import com.google.i18n.phonenumbers.NumberParseException;
+
+import java.util.Calendar;
+
+public class SupervisorSmsListener extends BroadcastReceiver {
+
+    private static final String SMS_RECEIVED = "android.provider.Telephony.SMS_RECEIVED";
+    private static final int THIRTY_MINUTES = 1800;
+    private String scheduleId;
+
+    IsSmsSenderACurrentEventSupervisor isSmsSenderACurrentEventSupervisor = new IsSmsSenderACurrentEventSupervisor();
+
+    public void onReceive(Context context, Intent receiverIntent) {
+        if (receiverIntent.getAction().equals(SMS_RECEIVED)) {
+            Bundle bundle = receiverIntent.getExtras();
+            if (bundle != null) {
+                // get sms objects
+                Object[] pdus = (Object[]) bundle.get("pdus");
+                if (pdus.length == 0) {
+                    return;
+                }
+                // large message might be broken into many
+                SmsMessage[] messages = new SmsMessage[pdus.length];
+                StringBuilder sb = new StringBuilder();
+                for (int i = 0; i < pdus.length; i++) {
+                    messages[i] = SmsMessage.createFromPdu((byte[]) pdus[i]);
+                    sb.append(messages[i].getMessageBody());
+                }
+                String smsSenderNum = messages[0].getOriginatingAddress();
+                scheduleId = sb.toString();
+                Timestamp currentTime = new Timestamp(Calendar.getInstance().getTime());
+                String formattedSmsSenderNum;
+                try {
+                    formattedSmsSenderNum = PhoneNumberFormatter.formatNumber(smsSenderNum,context);
+                } catch (NumberParseException e) {
+                    e.printStackTrace();
+                    Toast.makeText(context, "FAILED to format phone number. Process failed", Toast.LENGTH_LONG).show();
+                    return;
```

Comment:
You shouldn't return if the catch is called, as it prevents the second method from happening. As an example, on my device it went to  this catch, but I would have passed the second method if the return was simply removed.

I also think the toast should be removed as it isn't relevant to the user, and it would be weird if it popped out while the user is doing something else.

Reply:
From your comment, I realized you where talking about the return in the AuthenticatorByPhoneNumber class instead of the SupervisorSmsListener class. So I made the changes

Reply:
Good catch!